### ISSUE TITLE
Cannot run spec for a plugin targeting Logstash 1.x

### ISSUE THREAD

**Issue by wiibaa** [elastic/logstash#4188]
Trying to publish a plugin for Logstash 1.x, I'm facing this issue on my own plugin but it is general to all plugin with a core-api-v1 branch (using the date plugin as example below)

### Steps to reproduce

I'm using a fresh jruby install (through rvm)

    git clone https://github.com/logstash-plugins/logstash-filter-date.git
    cd logstash-filter-date
    checkout core-api-v1
    bundle install
    bundle exec rspec
   
And :boom: 

    RuntimeError: Logstash expects concurrent-ruby version 0.9.1 and version 0.9.2 is installed, please verify this patch: /home/philippe/.rvm/gems/jruby-1.7.22/gems/logstash-core-1.5.5-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb

Any idea ?

**Comment by purbon** [comment]
@wiibaa yea, this is a problem. thanks man! if you want to fix it right now, please just fix the concurrent-ruby to 0.9.1 in your gemspec. we'll be working on some bugfix release soon.

**Comment by purbon** [comment]
I tested this with the twitter input and I see the same error

```
skywalker% bundle exec rspec
RuntimeError: Logstash expects concurrent-ruby version 0.9.1 and version 0.9.2 is installed, please verify this patch: /Users/purbon/.rbenv/versions/jruby-1.7.20/lib/ruby/gems/shared/gems/logstash-core-1.5.5-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb
           (root) at /Users/purbon/.rbenv/versions/jruby-1.7.20/lib/ruby/gems/shared/gems/logstash-core-1.5.5-java/lib/logstash/patches/silence_concurrent_ruby_warning.rb:53
          require at org/jruby/RubyKernel.java:1072
           (roo
```

**Comment by purbon** [comment]
Related to #4173

**Comment by colinsurprenant** [comment]
@wiibaa until we figure if we push a new logstash-core gem version for this, you can also just edit the `Gemfile` and add 
```ruby
gem "concurrent-ruby", "0.9.1"
``` 
and redo
```
bundle install
```
that should fix it for now.

**Comment by wiibaa** [comment]
@purbon @colinsurprenant thanks for the quick reply, work like a charm. A side question about gemspec, even though the gemspec is using the version string `"logstash-core", '>= 1.4.0', '< 2.0.0'
 `, bundle install is "sometimes" (mainly after switching between v1 and v2 plugins) reporting `Using logstash-core 2.0.0.rc1`  any idea why ?

**Comment by purbon** [comment]
@wiibaa probably stale lock file? _(gessing)_

**Comment by purbon** [comment]
@wiibaa this is because the requirement is broken, you should change the expression to be ```s.add_runtime_dependency "logstash-core", '~> 1.4'  ```

**Comment by ph** [comment]
Concerning `concurrent-ruby` we will release a new version of logstash-core 1.5.6 which will fix that problem and correctly fix it to 0.9.2.

For the problem concerning the definition of the dependency on 1.5 we didn't find a correct solution yet and we are still investigating.

**Comment by colinsurprenant** [comment]
@wiibaa here's the state of our reflexion on this in #4236 

For now, in development mode, when working on a local plugin source, the best thing to do is the fix the version constrains in the Gemfile and not touch the gemspec. Here, the best thing is to also add logstash-core in the Gemfile.

```
gem "concurrent-ruby", "0.9.1"
gem "logstash-core", "~> 1.4"
```

So this solves the problem for the development environment and for the release/publishing the gemspec does not need to be updated.
